activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1829,6 +1839,35 @@ def check_if_method_has_arguments!(method_name, args, message = nil)
         end
       end
 
+      def process_select_args(fields)
+        fields.flat_map do |field|
+          if field.is_a?(Hash)
+            transform_select_hash_values(field)
+          else
+            field
+          end
+        end
+      end
+
+      def transform_select_hash_values(fields)
+        fields.flat_map do |key, columns_aliases|
+          case columns_aliases
+          when Hash
+            columns_aliases.map do |column, column_alias|
+              arel_column("#{key}.#{column}", &:itself).as(column_alias.to_s)
+            end
+          when Array
+            columns_aliases.map do |column|
+              arel_column("#{key}.#{column}", &:itselft)
```

Comment:
```suggestion
              arel_column("#{key}.#{column}", &:itself)
```
typo here, that suggest this path isn't covered by tests.